consensus/parlia/parlia.go

```diff
@@ -874,6 +885,22 @@ func (p *Parlia) Seal(chain consensus.ChainHeaderReader, block *types.Block, res
 	return nil
 }
 
+func (p *Parlia) shouldWaitForCurrentBlockProcess(chain consensus.ChainHeaderReader, header *types.Header, snap *Snapshot) bool {
+	if snap.inturn(p.val) {
```

Comment:
I think `snap`  is not needed here, we can use `if header.Difficulty == inturnDiff`